docs/manual/must-call-checker.tex

```diff
@@ -0,0 +1,204 @@
+
+%% This feature can be disabled by passing the \<-AnoLightweightOwnership> command-line parameter to the Must
+%% Call Checker.
+
+\sectionAndLabel{Assumptions about reflection}{must-call-reflection}
+
+The Must Call Checker is unsound with respect to reflection: it assumes
+that objects instantiated reflectively do not have must-call obligations (i.e. that their
+type is \<@MustCall({})>. If the checker is run with \<-AresolveReflection>, then
+this assumption applies only to types that cannot be resolved.
+
+\sectionAndLabel{Type parameter bounds often need to be annotated}{must-call-type-params}
+
+In an unannotated program, there may be mismatches between defaulted type
+qualifiers that lead to type-checking errors.  That is, the defaulted
+annotations at two different locations may be different and in conflict.  A
+specific example is in code that contains a mix of explicit upper bounds
+with an \<extends> clause and implicit upper bounds without an \<extends>
+clause.
+
+For example, consider the following example (from
+\href{https://github.com/plume-lib/plume-util}{plume-lib/plume-util}) of an
+interface with explicit upper bounds and a client with implicit upper
+bounds:
+\begin{Verbatim}
+  interface Partition<ELEMENT extends @Nullable Object, CLASS extends @Nullable Object> {}
+
+  class MultiRandSelector<T> {
+    private Partition<T, T> eq;
+  }
+\end{Verbatim}
+
+The code contains no \<@MustCall> annotations.
+Running the Must Call Checker on this code produces an error at each use
+of \<T> in \<MultiRandSelector>:
+
+\begin{smaller}
+\begin{Verbatim}
+error: [type.argument] incompatible type argument for type parameter ELEMENT of Partitioner.
+        private Partitioner<T, T> eq;
+                            ^
+  found   : T extends @MustCallUnknown Object
+  required: @MustCall Object
+error: [type.argument] incompatible type argument for type parameter CLASS of Partitioner.
+        private Partitioner<T, T> eq;
+                               ^
+  found   : T extends @MustCallUnknown Object
+  required: @MustCall Object
+\end{Verbatim}
+\end{smaller}
+
+The defaulted Must Call annotations differ.
+\<Partitioner> has an explicit bound, which uses the usual default of \<@MustCall(\{\})>.
+\<MultiRandSelector> has an implicit bound, which defaults to top (that is,
+\<@MustCallUnknown>), as explained in Section~\ref{climb-to-top}.
+
+In many cases, including this one, you can eliminate the false positive
+warning without writing any \<@MustCall> annotations.
+You can either:
+\begin{itemize}
+\item
+  adding an explicit bound to \<MultiRandSelector>, changing its declaration to
+  \code{class MultiRandSelector<T extends Object>}, or
+\item removing the explicit bound from \<Partition>, changing its declaration to
+  \code{interface Partition<ELEMENT, CLASS>}.
+\end{itemize}
+
+\noindent
+These two changes are semantically equivalent.  In this case, we preferred
```

Comment:
These changes are not semantically equivalent.  In the first bullet, `Partition<@MustCall("close") Object, Object>` is **not** a legal type, in the second bullet it is.

Reply:
Good catch. I'm not sure this paragraph adds much value anyway here, so I'm just going to remove it.